core/types/event.go

```diff
@@ -81,3 +81,12 @@ type DownlinkEventData struct {
 	GatewayID string                   `json:"gateway_id,omitempty"`
 	Config    *DownlinkEventConfigInfo `json:"config,omitempty"`
 }
+
+// DeviceEventData contains the updated of a set/updated device
+type DeviceEventData struct {
+	Latitude   float32           `json:"latitude,omitempty"`
+	Longitude  float32           `json:"longitude,omitempty"`
+	Altitude   int32             `json:"altitude,omitempty"`
+	Attributes map[string]string `json:"attributes,omitempty"`
```

Comment:
Not used?

Reply:
yes forgot to remove it. It was first included but the `changeList` function doesn't detect change in the map.